packages/core/src/Cardano/types/Governance.ts

```diff
@@ -0,0 +1,140 @@
+import { Lovelace } from './Value';
+import { ProtocolParametersUpdate } from './ProtocolParameters';
+
+/**
+ * Datum hash, this allows to specify a Datum without publicly revealing its value. To spend an output which specifies
+ * this type of datum, the actual Datum value must be provided and will be added to the witness set of
+ * the transaction.
+ */
+export type DataHash = Crypto.Hash32ByteBase16;
+
+export type Anchor = {
+  url: string;
+  dataHash: DataHash;
+};
+
+// Actions
+export enum GovernanceActionType {
+  parameter_change_action = 'parameter_change_action',
+  hard_fork_initiation_action = 'hard_fork_initiation_action',
+  treasury_withdrawals_action = 'treasury_withdrawals_action',
+  no_confidence = 'no_confidence',
+  new_committee = 'new_committee',
+  new_constitution = 'new_constitution',
+  info_action = 'info_action'
+}
+
+export type GovernanceActionId = {
+  id: Crypto.Hash32ByteBase16;
+  actionIndex: number;
+};
+
+export type CommitteeMember = {
+  coldCredential: Credential;
+  epoch: EpochNo;
+};
+
+export type Committee = {
+  members: [CommitteeMember];
+  quorumThreshold: Fraction;
+};
+
+export type Constitution = {
+  anchor: Anchor;
+  scriptHash: DataHash | null;
+};
+
+export type ParameterChangeAction = {
+  __typename: GovernanceActionType.parameter_change_action;
+  governanceActionId: GovernanceActionId | null;
+  protocolParamUpdate: ProtocolParametersUpdate;
+};
+
+export type HardForkInitiationAction = {
+  __typename: GovernanceActionType.hard_fork_initiation_action;
+  governanceActionId: GovernanceActionId | null;
+  protocolVersion: ProtocolParametersUpdate['protocolVersion'];
+};
+
+export type TreasuryWithdrawalsAction = {
+  __typename: GovernanceActionType.treasury_withdrawals_action;
+  withdrawals: Set<{
+    rewardAccount: RewardAccount;
+    coin: Lovelace;
+  }>;
+};
+
+export type NoConfidence = {
+  __typename: GovernanceActionType.no_confidence;
+  governanceActionId: GovernanceActionId | null;
+};
+
+export type NewCommittee = {
+  __typename: GovernanceActionType.new_committee;
+  governanceActionId: GovernanceActionId | null;
+  committeeColdCredentials: Set<Credential>;
+  committee: Committee;
+};
+
+export type NewConstitution = {
+  __typename: GovernanceActionType.new_constitution;
+  governanceActionId: GovernanceActionId | null;
+  constitution: Constitution;
+};
+
+export type InfoAction = {
+  __typename: GovernanceActionType.info_action;
+  message: string;
+};
+
+export type GovernanceAction =
+  | ParameterChangeAction
+  | HardForkInitiationAction
+  | TreasuryWithdrawalsAction
+  | NoConfidence
+  | NewCommittee
+  | NewConstitution
+  | InfoAction;
+
+export enum Vote {
+  no = 0,
+  yes = 1,
+  abstain = 2
+}
+
+export type VotingProcedure = {
+  vote: Vote;
+  anchor: Anchor | null;
+};
+
+export type VotePayload = {
+  governanceActionId: GovernanceAction;
+  votingProcedure: VotingProcedure;
+};
+
+export type Voter = Credential;
```

Comment:
`Voter` could be a keyhash or a script hash, but they could represent different things (e.g. CC keyhash, DRep keyhash or stakingPool keyhash). We'll need to provide a credential type that holds this information.
```
; Constitutional Committee Hot KeyHash: 0
; Constitutional Committee Hot ScriptHash: 1
; DRep KeyHash: 2
; DRep ScriptHash: 3
; StakingPool KeyHash: 4
voter =
  [ 0, addr_keyhash
  // 1, scripthash
  // 2, addr_keyhash
  // 3, scripthash
  // 4, addr_keyhash
  ]
```